File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

// get float for how much change is owed
// if neg # is given, ask again
float get_change(string prompt)
{
    float n;
    do
    {
       n = get_float("%s", prompt);
    }
    while (n < 0);
    return n;
}

//cycle through the coins deducting the changeAmt
int numCoins(int cents) {
   
     int q = cents/25;
     cents = cents-(25*q);
    
     int d = cents/10;
     cents = cents-(10*d);
    
     int n = cents/5;
     cents = cents-(5*n);
    
     int p = cents;
    
    return(q+d+n+p);
}

int main(void)
{
    //run get_change and make sure to get positive float
    float change = get_change("Change owed: ");
    //round changeAmt to dollars
    int myCents = round(change * 100);
    //run numCoins to cycle through quarters, dimes, nickels, and pennies
    int coins = numCoins(myCents);
    //print the number of coins we should get back
    printf("I would give you this many coins:\n");
    printf("%i\n", coins);
    
}
    

File: cash.py
from cs50 import get_float


def main():
    
    # run function to prompt user for change owed
    change = get_change()
    
    # muultiply change by 100 to calculate 
    myCents = round(change*100)
    
    # run number of coins function to calculate 
    coins = numCoins(myCents)
    
    print(coins)


def get_change():
    while True:
        # promt user for Change owed
        change = get_float("Change owed: ")
        if change > 0:
            break
    return change


def numCoins(cents):
    
    # quarters
    q = int(cents/25)
    cents = cents-(25*q) 

    # dimes
    d = int(cents/10)
    cents = cents-(10*d)

    # nickels
    n = int(cents/5)
    cents = cents-(5*n)
    
    # no need to calculate pennies, as all cents left are pennies
    # return change 
    return(q+d+n+cents)


main()
